bin/check_supp_models.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
+import sys
+import csv
+import argparse
+
+from Fred2.EpitopePrediction import EpitopePredictorFactory
+
+
+def convert_allele_back(allele):
+    if str(allele).startswith("H-2-"):
+        # convert internal Fred2 representation back to the nf-core/epitopeprediction input allele format
+        return allele.replace("H-2-", "H2-")
+    elif allele.startswith("HLA-"):
+        return allele.replace("HLA-", "")
+    else:
+        raise ValueError("Allele type unknown: " + allele + ". Currently expects allele to start either with 'HLA-' or 'H-2-'.")
+
+
+def __main__():
+    parser = argparse.ArgumentParser("Write out information about supported models by Fred2 for installed predictor tool versions.")
+    parser.add_argument('-v', '--versions', help='File with used software versions.', required=True)
+    args = parser.parse_args()
+
+    # NOTE this needs to be updated manually, if other methods should be used in the future
+    available_methods = ['syfpeithi', 'mhcflurry', 'mhcnuggets-class-1', 'mhcnuggets-class-2']
+    with open(args.versions, 'r') as versions_file:
+        tool_version = [ (row[0], str(row[1][1:])) for row in csv.reader(versions_file, delimiter = "\t") ]
+        # NOTE this needs to be updated, if a newer version will be available via Fred2 and should be used in the future
+        tool_version.append(('syfpeithi', '1.0'))
+        # get for each method the corresponding tool version
+        methods = { method:version for tool, version in tool_version for method in available_methods if tool.lower() in method.lower() }
+
+    for method, version in methods.items():
+        if version not in EpitopePredictorFactory.available_methods()[method]:
+            raise ValueError("The specified version " + version + " for " + method + " is not supported by Fred2.")
+
+        predictor = EpitopePredictorFactory(method, version=version)
+        with open(method + ".v" + str(version) + ".supported_alleles.txt", 'w') as output:
```

Comment:
print it additionally?

Reply:
you mean with the nextflow log.info? I planned that originally, but it is a lot of information, so one would probably end up redirecting that anyway, and in this way one at least has the information in a structured way. Or do you think that's not necessary?

Reply:
As discussed, let's just print a comment to command line (from `main.nf` I guess) where the files can be found.